src/target/nrf91.c

```diff
@@ -3,18 +3,75 @@
 #include "target_internal.h"
 #include "cortexm.h"
 #include "adiv5.h"
+#include "gdb_packet.h"
 
 /* Non-Volatile Memory Controller (NVMC) Registers */
-#define NRF91_NVMC          0x50039000U
-#define NRF91_NVMC_READY    (NRF91_NVMC + 0x400U)
-#define NRF91_NVMC_CONFIG   (NRF91_NVMC + 0x504U)
-#define NRF91_NVMC_ERASEALL (NRF91_NVMC + 0x50cU)
+#define NRF91_NVMC           0x50039000U
+#define NRF91_NVMC_READY     (NRF91_NVMC + 0x400U)
+#define NRF91_NVMC_READYNEXT (NRF91_NVMC + 0x408U)
+#define NRF91_NVMC_CONFIG    (NRF91_NVMC + 0x504U)
+#define NRF91_NVMC_ERASEALL  (NRF91_NVMC + 0x50cU)
+
+#define NVMC_TIMEOUT_MS 300U
 
 #define NRF91_NVMC_CONFIG_REN  0x0U // Read only access
 #define NRF91_NVMC_CONFIG_WEN  0x1U // Write enable
 #define NRF91_NVMC_CONFIG_EEN  0x2U // Erase enable
 #define NRF91_NVMC_CONFIG_PEEN 0x3U // Partial erase enable
 
+/* https://infocenter.nordicsemi.com/topic/ps_nrf9160/dif.html */
+#define NRF91_PARTNO 0x90U
+
+#define NRF91_CTRL_AP_RESET          ADIV5_AP_REG(0x000)
+#define NRF91_CTRL_AP_ERASEALL       ADIV5_AP_REG(0x004)
+#define NRF91_CTRL_IDR_EXPECTED      0x12880000
+#define NRF91_AHB_AP_IDR_EXPECTED    0x84770001
```

Comment:
These are missing their `U` suffixes to ensure they're defined as unsigned values. Because the AP IDR value (as an example) sets the sign bit on 32-bit platforms, if this value is then widened for any reason, then the upper bytes of the new value will be all set high, and if it is narrowed, this invokes UB (due to changing the value of the sign bit). Neither of these problems occur with unsigned constants.